atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/persistentlock/LockEntry.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.persistentlock;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.SortedMap;
+
+import org.apache.commons.lang3.StringUtils;
+import org.immutables.value.Value;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+
+@Value.Immutable
+public abstract class LockEntry {
```

Comment:
If we had an AtlasSchema, this may be a nice place to put this table. Do we want to introduce an AtlasSchema now? If so, do we want to add all the other "always there"/"hidden" tables (timestamp, transaction, etc) to that schema? What will the story look like in terms of migrations?